homeassistant/components/plugwise/climate.py

```diff
@@ -1,227 +1,214 @@
+        self._unique_id = f"cl-{dev_id}-{self._name}"
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._unique_id
 
     @property
     def hvac_action(self):
-        """Return the current hvac action."""
-        if self._heating_status or self._boiler_status or self._dhw_status:
-            return CURRENT_HVAC_HEAT
-        if self._cooling_status:
-            return CURRENT_HVAC_COOL
+        """Return the current action."""
+        if self._single_thermostat:
+            if self._heating_state or self._boiler_state:
+                return CURRENT_HVAC_HEAT
+            if self._cooling_state:
+                return CURRENT_HVAC_COOL
+            return CURRENT_HVAC_IDLE
+        if self._heating_state is not None or self._boiler_state is not None:
+            if self._setpoint > self._temperature:
+                return CURRENT_HVAC_HEAT
+            if self._cooling_state:
+                return CURRENT_HVAC_COOL
+            return CURRENT_HVAC_IDLE
+        if self._heating_state is not None or self._boiler_state is not None:
+            if self._setpoint > self._temperature:
+                return CURRENT_HVAC_HEAT
         return CURRENT_HVAC_IDLE
 
     @property
     def name(self):
         """Return the name of the thermostat, if any."""
         return self._name
 
+    @property
+    def device_info(self) -> Dict[str, any]:
+        """Return the device information."""
+
+        via_device = self._api.gateway_id
+        if self._dev_id is via_device:
+            via_device = None
+
+        return {
+            "identifiers": {(DOMAIN, self._dev_id)},
+            "name": self._name,
+            "manufacturer": "Plugwise",
+            "model": self._model.replace("_", " ").title(),
+            "via_device": via_device,
+        }
+
     @property
     def icon(self):
         """Return the icon to use in the frontend."""
-        return DEFAULT_ICON
+        return THERMOSTAT_ICON
 
     @property
     def supported_features(self):
         """Return the list of supported features."""
         return SUPPORT_FLAGS
 
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
     @property
     def device_state_attributes(self):
         """Return the device specific state attributes."""
         attributes = {}
-        if self._outdoor_temperature:
-            attributes["outdoor_temperature"] = self._outdoor_temperature
         if self._schema_names:
             attributes["available_schemas"] = self._schema_names
         if self._selected_schema:
             attributes["selected_schema"] = self._selected_schema
-        if self._boiler_temperature:
-            attributes["boiler_temperature"] = self._boiler_temperature
-        if self._water_pressure:
-            attributes["water_pressure"] = self._water_pressure
```

Comment:
Maybe we should not remove these until we add the sensor-platform?

Reply:
@bdraco already included merge notes on this (in bold at top of the PR) so I guess we don't have to worry about it.

Reply:
Ok, clear, then we leave them in.